### PR TITLE
fix crash in FontCollection::init() when FontFamily is empty

### PR DESCRIPTION
**Pull request by liguisheng**
Fixes：[flutter#72062] (flutter/flutter#72062)

### CODE REVIEW

File: third_party/txt/src/txt/paragraph_txt.cc [hunk 1]
```diff
@@ -807,6 +807,14 @@ void ParagraphTxt::Layout(double width) {
 
       std::shared_ptr<minikin::FontCollection> minikin_font_collection =
           GetMinikinFontCollectionForStyle(run.style());
+      if (minikin_font_collection == nullptr) {
```
Comment (jason-simmons) [inline third_party/txt/src/txt/paragraph_txt.cc:810]: Remove these logs.  The logging of the errors in `minikin::FontCollection::init` is sufficient.

**Review by jason-simmons — CHANGES_REQUESTED** [review]

File: third_party/txt/src/minikin/FontCollection.cpp [hunk 2]
```diff
@@ -91,10 +82,13 @@ void FontCollection::init(
     mSupportedAxes.insert(supportedAxes.begin(), supportedAxes.end());
   }
   nTypefaces = mFamilies.size();
-  LOG_ALWAYS_FATAL_IF(nTypefaces == 0,
-                      "Font collection must have at least one valid typeface");
-  LOG_ALWAYS_FATAL_IF(nTypefaces > 254,
-                      "Font collection may only have up to 254 font families.");
+  if (nTypefaces == 0 || nTypefaces > 254) {
```
[the remark this answers is not in the document]
Comment (liguisheng) [reply third_party/txt/src/minikin/FontCollection.cpp]: ok

File: third_party/txt/src/minikin/FontCollection.h [hunk 3]
```diff
@@ -29,9 +29,7 @@ namespace minikin {
 
 class FontCollection {
  public:
-  explicit FontCollection(
-      const std::vector<std::shared_ptr<FontFamily>>& typefaces);
-  explicit FontCollection(std::shared_ptr<FontFamily>&& typeface);
+  explicit FontCollection();
```
[the remark this answers is not in the document]
Comment (liguisheng) [reply third_party/txt/src/minikin/FontCollection.h:32]: ok

File: third_party/txt/src/minikin/FontCollection.h [hunk 4]
```diff
@@ -78,6 +76,9 @@ class FontCollection {
     mFallbackFontProvider = std::move(ffp);
   }
 
+  // Initialize the FontCollection.
+  bool init(const std::vector<std::shared_ptr<FontFamily>>& typefaces);
```
[the remark this answers is not in the document]
Comment (liguisheng) [reply third_party/txt/src/minikin/FontCollection.h]: ok

File: third_party/txt/src/txt/paragraph_txt.cc (hunk 1 above)
Comment (liguisheng) [reply third_party/txt/src/txt/paragraph_txt.cc:810]: done

**Review by liguisheng — COMMENTED** [review]
done

**Comment by jason-simmons** [comment]
Flutter runs a code format checker script in the GitHub presubmit checks.  Please apply the patches generated by the script (see https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8860965572236330288/+/steps/format_and_dart_test/0/stdout)

You can also run the formatter script locally (https://github.com/flutter/engine/blob/master/ci/format.sh)

**Comment by liguisheng** [comment]
@jason-simmons   all checks have passed

**Review by jason-simmons — APPROVED** [review]

**Pull request merged**